### DISCUSSION TITLE
Midgame

### DISCUSSION THREAD

**Discussion by amylizzle** [stonepillars/goon-flock#7] [Ideas]
I think we're all relatively agreed that the building and defending of the relay is the endgame, but we ought to discuss what we intend for the midgame.

To my mind, midgame is the point at which the flock has likely been discovered by the crew and has reached sufficient numbers that it's a battle to control. The goal of the flock at this point is to build up the resource required to build the relay (be that compute/power/gnesis/whatever) by harvesting the station and its resources using infrastructure. The crew obviously works to prevent this, by attacking infrastructure and impeding the drones.

It's already partially coded that the "Collector" structure amasses "power" from the number of tiles its connected to, with hints that the original author wanted it to leech from nearby APCs and cables. We could continue in that direction. We could also re-jig it to be compute power, and have the drones and bits contribute to that also. Maybe it should take a certain amount of compute before you can construct the relay? Maybe the relay can only be activated when you have x amount of compute?

Anyways, this is going to be the bulk of the gameplay - so we should get it right. Discuss.

**Comment by stonepillars** [comment]
I think the midgame should focus partially on flocktraces, but also on wider-scale logistics; expanding your power and collecting resources from the crew. Maybe the flockmind should start to reclaim parts of the station it doesn't need (unnecessary walls/rooms/machines) when loose items run out or get expanded building ability in the midgame?

~~Drone upgrades? Doodads? More buildings?~~ (This may result in some scope creep; should stick to the fundamentals of the midgame first)

**Comment by amylizzle** [comment]
Following some discussion on discord, we settled on Compute as the resource to gather to reach the end game.
Other things compute does:

- More compute lets you have more flocktraces
- Drones provide a small amount
- You can build dedicated structures which provide large chunks of compute
- You can capture human-built computers to provide medium amounts of compute
- You can capture human-built mainframes to provide very large chunk of compute
- Once you've reached the threshold, you can build the relay (maybe the relay should stop charging if you drop below the threshold again?)
